src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/MarkupTransitionCompletionItemProvider.cs

```diff
@@ -13,26 +13,17 @@ namespace Microsoft.CodeAnalysis.Razor.Completion;
 
 internal class MarkupTransitionCompletionItemProvider : IRazorCompletionItemProvider
 {
-    private static readonly ImmutableArray<RazorCommitCharacter> s_elementCommitCharacters = RazorCommitCharacter.CreateArray([">"]);
-
     private static RazorCompletionItem? s_markupTransitionCompletionItem;
+
     public static RazorCompletionItem MarkupTransitionCompletionItem
     {
         get
         {
-            if (s_markupTransitionCompletionItem is null)
-            {
-                var completionDisplayText = SyntaxConstants.TextTagName;
-                s_markupTransitionCompletionItem = new RazorCompletionItem(
-                    completionDisplayText,
-                    completionDisplayText,
-                    RazorCompletionItemKind.MarkupTransition,
-                    commitCharacters: s_elementCommitCharacters);
-                var completionDescription = new MarkupTransitionCompletionDescription(CodeAnalysisResources.MarkupTransition_Description);
-                s_markupTransitionCompletionItem.SetMarkupTransitionCompletionDescription(completionDescription);
-            }
-
-            return s_markupTransitionCompletionItem;
+            return s_markupTransitionCompletionItem ??= RazorCompletionItem.CreateMarkupTransition(
+                displayText: SyntaxConstants.TextTagName,
+                insertText: SyntaxConstants.TextTagName,
+                descriptionInfo: new(CodeAnalysisResources.MarkupTransition_Description),
+                commitCharacters: RazorCommitCharacter.CreateArray([">"]));
         }
```

Comment:
Nit: I that we replaced these checks with ArgHelper.ThrowIfNull in some cases but not in others. Curious about reasoning behind the choices, e.g. why not use ArgHelper.ThrowIfNull here?

Reply:
IMO, there isn't a need to use `ArgHelper.ThrowIfNull` in internal code called exclusively by our code if the calls won't compile when null is passed.